src/main/java/com/snowplowanalytics/redash/RedashClient.java

```diff
@@ -0,0 +1,437 @@
+        String response = post(url, new Gson().toJson(dataSource), CheckResponseStatus.NO);
+        int id = getIdFromJson(response);
+        dataSource.setId(id);
+        return id;
+    }
+
+    /**
+     * Updates data-source on Redash server with provided as argument DataSource instance. Please note that password field
+     * doesn't take a part in making decision of necessity to update data-source as the server does not return it (password field).
+     *
+     * @param dataSource Argument's name field is used to find specific data-source on the server which (if found) then
+     *                   will be updated by it's fields, if it is necessary.
+     * @return false, if all the fields except password are equals, and there is nothing to update.
+     * True if one or more fields needs to be updated.
+     * @throws IOException with the message that the server response in the case if data-source with provided name
+     *                     couldn't be found, or wrong API key, or there is any connection error.
+     */
+    public boolean updateDataSource(DataSource dataSource) throws IOException {
+        DataSource fromDataBase;
+        try {
+            fromDataBase = getDataSource(dataSource.getName());
+        } catch (IllegalArgumentException e) {
+            throw new IOException(e.getMessage());
+        }
+        if (isDataSourceAlreadyUpToDate(fromDataBase, dataSource)) {
+            return false;
+        }
+        String url = baseUrl + DATA_SOURCES_URL_PREFIX + "/" + fromDataBase.getId() + API_KEY_URL_PARAM + apiKey;
+        post(url, new Gson().toJson(dataSource), CheckResponseStatus.NO);
+        return true;
+    }
+
+    /**
+     * @return ArrayList<DataSource> of all existing data-sources on the server. If there are no data-sources then the list
+     * will be empty.
+     * @throws IOException if server is not available due to connection error or API key is invalid.
+     */
+    public List<DataSource> getDataSources() throws IOException {
+        String url = baseUrl + DATA_SOURCES_URL_PREFIX + API_KEY_URL_PARAM + apiKey;
+        Type listType = new TypeToken<ArrayList<DataSource>>() {
+        }.getType();
+        return new Gson().fromJson(get(url, CheckResponseStatus.YES), listType);
+    }
+
+    /**
+     * Removes data-source
+     *
+     * @param dataSourceId id of data-source which will be deleted if it exists.
+     * @return false if data-source with provided id doesn't exist on the server, if it does then data-source will be
+     * deleted and true will be returned.
+     * @throws IOException if server is not available due to connection error or API key is invalid.
+     */
+    public boolean deleteDataSource(int dataSourceId) throws IOException {
+        String url = baseUrl + DATA_SOURCES_URL_PREFIX + "/" + dataSourceId + API_KEY_URL_PARAM + apiKey;
+        return delete(url, CheckResponseStatus.NO).isEmpty();
```

Comment:
How is this asserting the removal of the data-source?  It looks like we are only checking for an empty string as a way of determining that it has been removed?